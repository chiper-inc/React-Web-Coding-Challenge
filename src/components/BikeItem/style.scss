.content-bike {
  &:hover{
      background-color: #2600f8;
    }
  @apply p-4
    mt-8 
    grid 
    grid-cols-1 
    md:grid-cols-bikeItem 
    md:items-center 
    gap-4
    border-2
    border-gray-400
    rounded;
    background-color: rgb(0, 255, 234);
    
  .content-img {
    @apply w-full h-full md:w-32 md:h-32;
    img {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  }

  .content-info {
    @apply flex flex-col justify-between gap-4;

    .title {
      &:hover{
        @apply text-white;
      }
      @apply text-2xl md:text-4xl text-blue-600 underline cursor-pointer;
      
    }
    .description {
      @apply text-base md:text-2xl text-gray-400;
    }

    .location {
      @apply text-base md:text-2xl text-red-600;
    }
  }
}
