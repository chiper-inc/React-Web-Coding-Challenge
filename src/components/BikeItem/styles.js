import styled from 'styled-components'

export const Wrapper = styled.div`
  display: grid;
  grid-auto-flow: column;
  grid-template-columns: .5fr 1fr;
  grid-gap: 2em;
  background: linear-gradient(85deg, #434343, #262626);
  color: #b5b5b5;
  padding: 2rem;
  border-radius: 16px;
  &:hover {
    opacity: 0.8;
    cursor: pointer;
  }
`;

export const ContainerImg = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  .bike-img {
    max-width: 100%;
  }
`;

export const ContainerInfo = styled.div`
  display: flex;
  flex-direction: column;
  gap: 6px;
`;

export const Description = styled.span`
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
`;
