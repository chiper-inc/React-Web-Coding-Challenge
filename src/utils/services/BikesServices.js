/* eslint-disable import/no-anonymous-default-export */
import axios from 'axios';

const URL = 'https://bikeindex.org:443/api';

const index = async (page = 1) => {
  try {
    const res = await axios.get(
      `${URL}/v3/search?page=${page}&per_page=10&location=berlin&stolenness=proximity`
    );
    // console.log('index response', res)
    return res.data;
  } catch (e) {
    throw e;
  }
};

const searchForTitle = async (title) => {
  try {
    const res = await axios.get(
      `${URL}/v3/search?per_page=10&manufacturer=${title}&location=berlin&stolenness=proximity`
    );
    // console.log('index response', res)
    return res.data;
  } catch (e) {
    throw e;
  }
};

const searchID = async (id) => {
  try {
    const res = await axios.get(`${URL}/v3/bikes/${id}`);
    // console.log('searchID response', res);
    return res.data;
  } catch (e) {
    throw e;
  }
};

export default { index, searchID, searchForTitle };
