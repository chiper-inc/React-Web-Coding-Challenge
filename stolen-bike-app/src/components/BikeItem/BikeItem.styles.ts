import styled from 'styled-components';

type Props = {
    isStolen: Boolean;
}

export const Card = styled.div `
    position: relative;
    z-index: 1;

    display: grid;
    grid-template-columns: 30% 50%;
    grid-template-rows: 100%;
    grid-column-gap: 1rem;
    grid-template-areas: "Image  InformationWrap";

    word-wrap: break-word;
    background-color: #fff;
    background-clip: border-box;
    border: 1px solid var(--gray);
    border-radius: 1rem;
    margin: .5rem 0;
    padding: .8rem 0;
`;

export const Image = styled.img `
    width: 8rem;
    height: 8rem;
    display: flex;
    justify-content: center;
    align-self: center;
    margin: auto;
    border-radius: 2rem;
    object-fit: cover;
    border: .4rem double var(--gray);
`

export const InformationWrap = styled.div `
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: center;
`

export const Information = styled.p `
    margin: 0;
`

export const Badge = styled.div <Props> `
    display: inline-block;
    padding: 0.35em 0.65em;
    margin: .5rem 0;
    font-size: .75em;
    font-weight: 700;
    line-height: 1;
    color: #fff;
    text-align: center;
    white-space: nowrap;
    vertical-align: baseline;
    border-radius: 0.25rem;
    background-color: ${(props) => props.isStolen ? '#dc3545' : '#0fa64b'};
`

