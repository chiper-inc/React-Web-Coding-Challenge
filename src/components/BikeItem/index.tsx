import { useNavigate } from "react-router-dom";
import { useFormatDate } from "../../hooks/useFormatDate";
import bike from "../../assets/images/bike.svg";
import { Wrapper, ContainerInfo, ContainerImg, Description } from "./styles";

interface IBikeItemProps {
  id: number;
  thumb: string;
  title: string;
  description: string;
  date_stolen: number;
  stolen_location: string;
}

export const BikeItem = ({
  id, thumb, title, description, date_stolen, stolen_location
}: IBikeItemProps) => {

  const navigate = useNavigate();
  const [date] = useFormatDate(date_stolen);

  const handleNavigate = () => navigate(`/case/${id}`);

  return (
    <Wrapper onClick={handleNavigate} >
      <ContainerImg>
        <img className="bike-img" src={thumb || bike} alt={title} />
      </ContainerImg>
      <ContainerInfo>
        <span><b>Title:</b> {title || 'No title'}</span>
        <Description><b>Description:</b> {description || 'No description'}</Description>
        <span><b>Date Stolen:</b> {date || 'Unknown'}</span>
        <span><b>Stolen Location:</b> {stolen_location || 'Unknown'}</span>
      </ContainerInfo>
    </Wrapper>
  );
};
