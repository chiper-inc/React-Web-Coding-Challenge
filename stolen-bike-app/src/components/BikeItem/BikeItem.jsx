import { Card, InformationWrap, Image, Information, Badge } from './BikeItem.styles';
import bike_not_found from './../../assets/img/bike_not_found.png';
import { Link } from 'react-router-dom';

export const BikeItem = ({bike}) => {

    const formatterDate = (unixDate) => {
        const date = new Date(parseInt(unixDate)*1000);
        const formatterDate = date.toDateString();
        return formatterDate;
    }

    return (
        <Card>
            <Image src={bike.thumb ? bike.thumb : bike_not_found} alt={bike.title} />
            <InformationWrap>
                <Link to={`bike/${bike.id}`}>
                    <h2>{bike.title}</h2>
                </Link>
                <Information>{bike.description ? bike.description : 'No Description'}</Information>
                <Information>{bike.date_stolen ? formatterDate(bike.date_stolen) : 'N/A'} - {bike.stolen_location}</Information>
                <Badge isStolen={bike.stolen}>{bike.stolen ? "Stolen" : "Recovered"}</Badge>
            </InformationWrap>
        </Card>
    )
}
