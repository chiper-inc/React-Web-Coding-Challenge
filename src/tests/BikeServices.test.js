import React from 'react';
import { shallow } from 'enzyme';
import BikeItem from '../components/BikeItem';
import BikesServices from '../utils/services/BikesServices';

test('Checking that the index returns 10 items per page', async () => {
  const bikesResponse = await BikesServices.index(1);
  await expect(bikesResponse.bikes.length).toEqual(10);
});

test('Checking that the title of the consulted case exists', async () => {
  const bikesResponse = await BikesServices.searchForTitle('surly');
  await expect(bikesResponse.manufacturer).toEqual('surly');
});
