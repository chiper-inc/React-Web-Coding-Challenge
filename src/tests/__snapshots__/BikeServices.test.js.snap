// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component Bike 1`] = `
<div
  className="content-bike "
>
  <div
    className="content-img"
  >
    <img
      src="noImage.png"
    />
  </div>
  <div
    className="content-info"
  >
    <Link
      className="title"
      to="/details/undefined"
    />
    <p
      className="description"
    >
      No description
    </p>
    <p
      className="location"
    >
      No location
    </p>
  </div>
</div>
`;
